### PR TITLE
Asg frontend v18 standalone

### PR DESCRIPTION
**Pull request by Mil-m**

### CODE REVIEW

**Comment by yroochun** [comment]
This PR is a basic upgrade to Angular 18, with most changes being automatic, either generated by running the upgrade command or made by the IDE during file reformatting. 

Even with minimal testing, multiple errors were found:

1. Data Portal component - Applying a filter (Annotation-Complete + Assemblies Submitted), then clicking on the main Data Portal link:
Issue: The filters remain applied. The page should refresh with the filters cleared.

2. Phylogeny Filters - The filters behave inconsistently. For example, applying a filter (assemblies done & symbionts submitted to biosamples), then clicking on ‘Other’:

Error: TypeError: Cannot read properties of undefined (reading 'key') in phylogeny-filter.component.html.

3. Several times, after filtering and selecting a phylogeny term (e.g., Metazoa), the phylogeny tree sometimes fails to load.

4. Clicking on the tree generates the following error:

zone.js:1188 Uncaught TypeError: record.tolid.charAt is not a function in d3tree.js.

5. The issue at http://localhost:4000/gis?phylogeny=class%20-%202203428 has already been raised.

Please ensure the application is thoroughly tested and all outstanding issues or bugs are resolved before submitting a PR

Screenshot 2024-10-02 at 19 16 48
Screenshot 2024-10-02 at 19 18 06
Screenshot 2024-10-03 at 08 23 06
Screenshot 2024-10-03 at 08 23 35
Screenshot 2024-10-03 at 08 24 18
Screenshot 2024-10-03 at 08 25 01

File: src/app/app.routes.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  { path: 'tracking/:id', component: TrackingDetailsComponent },
+  { path: 'tracking/details/:organism', component: TrackingDetailsComponent },
+  { path: 'tracking', component: TrackingSystemComponent },
+
+  { path: 'tree', component: PhylogeneticsComponent},
+  {
+    path: '',
+    children: [
+      { path: '404', component: NotFoundComponent },
+      { path: '**', component: NotFoundComponent }
```
Comment (Alexey-ebi) [inline src/app/app.routes.ts]: let's add titles here

**Comment by Alexey-ebi** [comment]
@yroochun thanks for bugs report, I can see that we have the same behaviour on prod at the moment as well, so it's not necessarily linked with current update. I would suggest to move current FE to v18 first and then work on these bugs.
Before that there are couple of issues with current code that we don't have on the prod:
1. http://localhost:4200/data/root/details/Acropora%20spathulata - compare it with https://portal.aquaticsymbiosisgenomics.org/data/root/details/Acropora%20spathulata. If table is too wide, on the prod I can scroll it to the right. With current code I can't do this. 
2. Same species, upon reload default tab is metadata and it should be data like on the prod now
3. http://localhost:4200/gis, search field must have same width as the map. It's now on the prod as well, but I guess it should be easy to fix?

File: src/app/dashboard/component/dashboard.component.ts [hunk 2]
```diff
@@ -1,28 +1,90 @@
...
+    MatFormFieldModule,
+    MatInputModule,
+    MatExpansionModule,
+    MatCheckboxModule,
+    MatTabsModule,
+    MatBadgeModule,
+    MatChipsModule,
+    MatListModule,
+    MatDialogModule
+  ],
```
Comment (Alexey-ebi) [inline src/app/dashboard/component/dashboard.component.ts]: Can you check that all these imports are really necessary please, I was able to compile code without any errors with just these imports:
MatPaginatorModule,
    MatTableModule,
    MatInputModule,
    NgxSpinnerModule,
    FilterComponent,
    PhylogenyFilterComponent,
    MatExpansionPanel,
    MatExpansionPanelHeader,
    FormsModule,
    MatCheckboxModule,
    ActiveFilterComponent,
    RouterLink,
    MatChipsModule,
    NgStyle,
    NgClass,
    MatSort

**Comment by yroochun** [comment]
@Alexey-ebi Alright, I'll create Jira tickets for these bugs.

On all pages, when the columns exceed the page size, they are hidden. I won’t proceed with the merge until this issue is resolved.

File: src/app/dashboard/component/details/details.component.html [hunk 3]
```diff
@@ -2,231 +2,309 @@
...
+                            <dt class="col-sm-3">Sample Derived From:</dt>
+                            <dd class="col-sm-9">
+                                <a href="http://www.ebi.ac.uk/biosamples/samples/{{bioSampleObj.sampleDerivedFrom}}" target="_blank">
+                                    {{bioSampleObj.sampleDerivedFrom}}
+                                </a>
+                            </dd>
+                        </ng-container>
+                    }
+                </ng-container>
+            }
```
Comment (Alexey-ebi) [inline src/app/dashboard/component/details/details.component.html]: You are not using "benefits" of new @if, etc... With @if there is no need to have a ng-container, so this code can be simplified to:
```
@if (bioSampleObj) {
                @if (bioSampleObj.commonName) {
                    <dt class="col-sm-3">Name:</dt>
                    <dd class="col-sm-9">{{bioSampleObj.commonName}}</dd>
                }
...
```

Comment (Alexey-ebi) [reply src/app/dashboard/component/details/details.component.html]: also can you test if you need to have nested conditions here, can we just use
```
@if (bioSampleObj?.commonName)
```

File: src/app/dashboard/component/details/details.component.ts [hunk 4]
```diff
@@ -1,53 +1,105 @@
...
+    RouterLink,
+    MatTable,
+    MatHeaderRow,
+    MatRow,
+    MatPaginator,
+    MatHeaderRowDef,
+    MatRowDef,
+    MatHeaderCellDef,
+    NgClass,
+    MatInputModule
```
Comment (Alexey-ebi) [inline src/app/dashboard/component/details/details.component.ts]: please check that all these imports are necessary

File: src/app/dashboard/component/organism-details/organism-details.component.html [hunk 5]
```diff
@@ -1,764 +1,716 @@
...
+                        <dt class="col-md-3">Tracking Status:</dt>
+                        <dd class="col-md-9">
+                            <mat-chip-set>
+                                <mat-chip [ngStyle]="{'background-color': 'gold'}">{{ bioSampleObj.currentStatus }}</mat-chip>
+                            </mat-chip-set>
+                        </dd>
+                    }
+                </ng-container>
+            }
         </dl>
```
Comment (Alexey-ebi) [inline src/app/dashboard/component/organism-details/organism-details.component.html]: same as for comments before, no need to have ng-container and nested ifs

File: src/app/dashboard/component/organism-details/organism-details.component.ts [hunk 6]
```diff
@@ -1,36 +1,61 @@
...
+    MatSortModule,
+    MapClusterComponent,
+    MatListModule,
+    RouterLink,
+    MatExpansionModule,
+    MatCheckboxModule,
+    FormsModule,
+    MatInputModule,
+    CommonModule
+  ],
```
Comment (Alexey-ebi) [inline src/app/dashboard/component/organism-details/organism-details.component.ts]: check that all these imports are required

File: src/app/dashboard/map-cluster/map-cluster.component.html [hunk 7]
```diff
@@ -1,5 +1,5 @@
-<div class="map-container">
-    <div class="map-frame">
-        <div id="map"></div>
+<div class="map-container" style="height: 100%; width: 100%;">
+    <div class="map-frame" style="height: 100%; width: 100%;">
+        <div #mapContainer style="height: 500px; width: 100%;"></div>
     </div>
```
Comment (Alexey-ebi) [inline src/app/dashboard/map-cluster/map-cluster.component.html]: would it be better to move all these styles into CSS

File: src/app/tracking-system/tracking-system/tracking-system.component.html [hunk 8]
```diff
@@ -21,45 +13,60 @@ <h3 class="card-header filter-heading">
...
+                                @for (filter of BiosamplesFilters; track filter) {
+                                    <ng-container>
+                                        @if (filter.key === 'Biosamples - done' || filter.key === 'Biosamples - Done') {
+                                            <li class="biosample list-group-item d-flex justify-content-between align-items-center biosamples-inactive" [ngClass]="checkFilterIsActive(filter.key)" (click)="onFilterClick($event, 'biosamples', filter.key)" style="cursor: pointer">
+                                                <span>Biosamples - Submitted</span>
+                                                <span class="badge badge-pill badge-info">{{filter.doc_count}}</span>
+                                            </li>
+                                        }
+                                    </ng-container>
+                                }
```
Comment (Alexey-ebi) [inline src/app/tracking-system/tracking-system/tracking-system.component.html]: no need to use nested ng-containers

File: src/app/tracking-system/tracking-system/tracking-system.component.html [hunk 9]
```diff
@@ -278,493 +289,789 @@ <h1 class="modal-title modal-heading" id="ontologyModal" modal-heading>Taxonomy
...
+                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    @if ((family.childData.length == 1 && familyChild.key != 'Other') || (family.childData.length > 1)) {
+
```
Comment (Alexey-ebi) [inline src/app/tracking-system/tracking-system/tracking-system.component.html]: what is that?

File: src/app/tracking-system/tracking-system/tracking-system.component.ts [hunk 10]
```diff
@@ -1,41 +1,86 @@
...
+    NgStyle,
+    MatHeaderRow,
+    MatHeaderRowDef,
+    MatRow,
+    MatRowDef,
+    MatPaginator,
+    UpperCasePipe,
+    MatInputModule,
+    MatColumnDef,
+    MatSortHeader
```
Comment (Alexey-ebi) [inline src/app/tracking-system/tracking-system/tracking-system.component.ts]: check that all imports are required here

**Review by Alexey-ebi — CHANGES_REQUESTED** [review]
Thanks Milena, I think that code in general starts to look a bit cleaner. I've added comments in couple of places but please make sure to check all code base for same problems:
1. All component imports are definitely required to build HTML page. I've tested it for one component and was able to remove huge amount if imports, please test it for all other components
2. When you are using new @if or @for, there is no need to have ng-container anymore, just use usuall html tags

File: src/app/tracking-system/tracking-system/tracking-system.component.html (hunk 9 above)
Comment (Mil-m) [reply src/app/tracking-system/tracking-system/tracking-system.component.html]: if you're saying about the size of html - it was such before...
should I rewrite this part in this PR?

**Comment by Raheelaaslam1024** [comment]
@Mil-m I reviewed the PR i think css is not working in this branch if you fix the css we can merge.
Screenshot 2024-10-24 at 20 04 18

**Comment by yroochun** [comment]
Sorting no longer displayed on the 'Organisms' page

**Comment by Mil-m** [comment]
> Sorting no longer displayed on the 'Organisms' page

What do you mean under 'sorting'?
I don't see any differences between my branch and https://portal.aquaticsymbiosisgenomics.org/data

**Comment by Mil-m** [comment]
> @Mil-m I reviewed the PR i think css is not working in this branch if you fix the css we can merge. Screenshot 2024-10-24 at 20 04 18

are you about this warning?

WARNING
/Users/mansurova/WebstormProjects/asg-frontend/src/app/dashboard/component/dashboard.component.ts depends on 'bootstrap'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.dev/tools/cli/build#configur

This page is displayed to me the same way as in production

**Pull request closed without merging**